ElectricCar.cpp

```diff
@@ -7,10 +7,21 @@ ElectricCar::ElectricCar(ElectricEngine* engine)
     std::cout << __FUNCTION__ << std::endl;
 }
 
-ElectricCar::~ElectricCar()       { std::cout << __FUNCTION__ << std::endl; }
-void ElectricCar::turnLeft()      { std::cout << __FUNCTION__ << std::endl; }
-void ElectricCar::turnRight()     { std::cout << __FUNCTION__ << std::endl; }
-void ElectricCar::brake()         { std::cout << __FUNCTION__ << std::endl; }
-void ElectricCar::accelerate(int) { std::cout << __FUNCTION__ << std::endl; }
+ElectricCar::~ElectricCar() { 
+    std::cout << __FUNCTION__ << std::endl;
+    delete engine_;
+}
 void ElectricCar::charge()        { std::cout << __FUNCTION__ << std::endl; }
 
+void ElectricCar::restore() {
+    charge();
+}
+
+void ElectricCar::changeEngine(int power, int batteryCapacity) {
```

Comment:
Isn't better to just take engine as parameter?

Reply:
Don't know if better but in different way.

Reply:
Passing engine would be a desired way. Then we can have something like EngineFactory that can produce engines and we can pass them to this function. However, for current needs it is ok.